subsys/bluetooth/host/keys.c

```diff
@@ -335,17 +364,31 @@ static int keys_set(const char *name, size_t len_rd, settings_read_cb read_cb,
 		BT_ERR("Failed to allocate keys for %s", bt_addr_le_str(&addr));
 		return -ENOMEM;
 	}
+	if (len != BT_KEYS_STORAGE_LEN) {
+#if defined(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
+		if (len == (BT_KEYS_STORAGE_LEN_COMPAT)) {
```

Comment:
```suggestion
		if (IS_DEFINED(CONFIG_BT_KEYS_OVERWIRET_OLDEST) &&
		    len == BT_KEYS_STORAGE_LEN_COMPAT) {
```

Reply:
We have no `BT_KEYS_STORAGE_LEN_COMPAT` defined when `CONFIG_BT_KEYS_OVERWIRET_OLDEST` not defined, it would not compile